framework/i18n/Formatter.php

```diff
@@ -787,6 +787,50 @@ public function asRelativeTime($value, $referenceTime = null)
         }
     }
 
+    /**
+     * Formats the value as a time interval.
+     * @param mixed $value the value to be formatted, in ISO8601 format.
+     * @return string the formatted result.
+     */
+    public function asInterval($value)
+    {
+        if ($value === null) {
+            return $this->nullDisplay;
+        }
+        if ($value instanceof \DateInterval) {
+            $negative = $value->invert;
+            $interval = $value;
+        } else {
+            if (strpos($value, '-') !== false) {
+                $negative = true;
+                $interval = new \DateInterval(str_replace('-', '', $value));
+            } else {
+                $negative = false;
+                $interval = new \DateInterval($value);
+            }
+        }
+        if ($interval->y > 1) {
+            $parts[] = Yii::t('yii', '{delta, plural, =1{a year} other{# years}}', ['delta' => $interval->y], $this->locale);
```

Comment:
I think you've copied `a year`, `a month`, `an hour`, etc from `Formatter::asRelevantTime`. This verbal form is used, when you are talking about a relevant time - `in an hour`, `a second ago`. In `asInterval` it's a wrong verbal form.
Replace all `a`, `an` with `#` and you will get correct `1 hour, 1 minute` or `1 day, 1 hour, 35 seconds` instead of `an hour, a minute, 31 seconds`